### ISSUE TITLE
Make NvimTreeFindFile find files outside of current working directory

### ISSUE THREAD

**Issue by mnowotnik** [kyazdani42/nvim-tree.lua#240]
NvimTreeFindFile doesn't find files that are outside of cwd. I've made a workaround, but imho it should be a feature:

```lua
local nvim_tree_find_file = function()
  local cwd = vim.fn.getcwd()
  local cur_path = vim.fn.expand('%:p:h')
  require('nvim-tree').refresh()
  require('nvim-tree.lib').change_dir(cur_path)
  require('nvim-tree').find_file(true)
  vim.cmd('cd ' .. cwd)
end
```

**Comment by kyazdani42** [comment]
duplicate of #141 but thanks for providing a workaround fix :)
The issue is i don't want to mess up the tree pwd if the file is in some ignored directory or cannot be found. I'll try to do something more robust with your solution !

**Comment by zealot128** [comment]
Thanks on @mnowotnik's snippet I made it work with a mapping:

```lua
_G.nvim_tree_find_file = function()
  local cwd = vim.fn.getcwd()
  local cur_path = vim.fn.expand('%:p:h')
  require('nvim-tree').refresh()
  require('nvim-tree.lib').change_dir(cur_path)
  require('nvim-tree').find_file(true)
  vim.cmd('cd ' .. cwd)
end

map("n", "-", ":lua _G.nvim_tree_find_file()<CR>")
```

Pressing "-" will open the Tree for the current file, even if the file is outside the current working dir. That's the most important feature I needed from https://github.com/tpope/vim-vinegar.

I am totally new to the whole Lua + Nvim story, so there might be at better way to configure.

**Comment by fuelen** [comment]
Here is my updates
```lua
_G.nvim_tree_find_file = function()
  local function starts_with(String, Start)
    return string.sub(String, 1, string.len(Start)) == Start
  end


  local cwd = vim.fn.getcwd()
  local cur_path = vim.fn.expand('%:p:h')

  if starts_with(cur_path, cwd) then
    require('nvim-tree').find_file(true)
  else
    require('nvim-tree').refresh()
    require('nvim-tree.lib').change_dir(cur_path)
    require('nvim-tree').find_file(true)
    vim.cmd('cd ' .. cwd)
  end
end
```
so it won't always change a directory, but only when it is not a child of the current one.

**Comment by jackieli-tes** [comment]
I believe the best approach is to provide a function that's able to accept any directory so that the user doesn't need to change working directory: Language servers usually working directory being current project, changing it will confuse the LSP. e.g. gopls

**Comment by kyazdani42** [comment]
fixed in 1.1.4 (66f46b2)
